django/chat/responses.py

```diff
@@ -457,18 +435,82 @@ async def add_files_to_library():
                 content_type="text/event-stream",
             )
 
+        # If we're stitching sources together into groups...
+        if chat.options.qa_granularity > 768:
```

Comment:
Note that the granularity resets to 768 in "combine sources" mode, so this block gets bypassed in that case